relayer/src/supervisor/spawn.rs

```diff
@@ -0,0 +1,544 @@
+                dst_client_id: client.client_id.clone(),
+                dst_chain_id: chain.id(),
+                src_chain_id: client.client_state.chain_id(),
+            });
+
+            self.workers
+                .spawn(
+                    client_object.clone(),
+                    counterparty_chain.clone(),
+                    chain.clone(),
+                )
+                .then(|| debug!("spawned Client worker: {}", client_object.short_name()));
+
+            // On config reload, we need to spawn the Client worker for the counterparty as well,
+            // otherwise we may never do it, eg. if the counterparty chain was not updated in the
+            // config.
+            if self.mode == SpawnMode::Reload {
+                let counterparty_client_id = connection.connection_end.counterparty().client_id();
+                let counterparty_client_object = Object::Client(Client {
+                    dst_client_id: counterparty_client_id.clone(),
+                    dst_chain_id: client.client_state.chain_id(),
+                    src_chain_id: chain.id(),
+                });
+
+                self.workers
+                    .spawn(
+                        counterparty_client_object.clone(),
+                        chain.clone(),
+                        counterparty_chain.clone(),
+                    )
+                    .then(|| {
+                        debug!(
+                            "spawned Client worker: {}",
+                            counterparty_client_object.short_name()
+                        )
+                    });
+            }
+
+            // TODO: Only start the Packet worker if there are outstanding packets or ACKs.
+            //       https://github.com/informalsystems/ibc-rs/issues/901
+
+            // create the path object and spawn worker
+            let path_object = Object::Packet(Packet {
+                dst_chain_id: counterparty_chain.id(),
+                src_chain_id: chain.id(),
+                src_channel_id: channel.channel_id,
+                src_port_id: channel.port_id,
+            });
+
+            self.workers
+                .spawn(
+                    path_object.clone(),
+                    chain.clone(),
+                    counterparty_chain.clone(),
+                )
+                .then(|| debug!("spawned Path worker: {}", path_object.short_name()));
+
+            if self.mode != SpawnMode::Reload {
+                return Ok(());
+            }
+
+            // On config reload, we need to spawn the Packet worker for the counterparty as well,
+            // otherwise we may never do it, eg. if the counterparty chain was not updated in the
+            // config.
+
+            let remote_channel = counterparty_channel.and_then(|c| {
+                c.remote
+                    .channel_id
+                    .as_ref()
+                    .map(|cid| (c.remote.port_id.clone(), cid.clone()))
+            });
```

Comment:
I think there is a problem here as the remote of the counterparty channel is the local one. We should just take the remote from channel. During testing the worker for direction from other chain to the added chain is not spawned.